plutus-tx/src/Language/PlutusTx/Evaluation.hs

```diff
@@ -31,43 +32,36 @@ import qualified Language.UntypedPlutusCore.Evaluation.Machine.Cek          as U
 import qualified Control.Exception
 import           System.IO.Unsafe
 
-stringBuiltins
-    :: ( GShow uni, GEq uni, uni `IncludesAll` '[String, Char, ()]
-       , Closed uni, uni `Everywhere` ExMemoryUsage
-       )
-    => DynamicBuiltinNameMeanings (CekValue uni)
-stringBuiltins =
-    insertDynamicBuiltinNameDefinition dynamicCharToStringDefinition
-        $ insertDynamicBuiltinNameDefinition dynamicAppendDefinition mempty
-
 -- | Evaluate a program in the CEK machine with the usual string dynamic builtins.
 evaluateCek
-    :: (GShow uni, GEq uni, DefaultUni <: uni, Closed uni, uni `Everywhere` ExMemoryUsage)
-    => Program Name uni () -> Either (CekEvaluationException uni) (Term Name uni ())
-evaluateCek (Program _ _ t) = UPLC.evaluateCek stringBuiltins PLC.defaultCostModel t
+    :: ( GShow uni, GEq uni, DefaultUni <: uni, Closed uni, uni `Everywhere` ExMemoryUsage
+       , fun ~ DefaultFun
+       )
+    => Program Name uni fun () -> Either (CekEvaluationException uni fun) (Term Name uni fun ())
+evaluateCek (Program _ _ t) = UPLC.evaluateCek defBuiltinsRuntime t
```

Comment:
@michaelpj we discussed if it should be `uni ~ PLC.DefaultUni` here, given that we have `fun ~ DefaultFun` anyway. I haven't ditched `DefaultUni <: uni` here, but I probably will as I think we both agree it makes sense to use a concrete universe whenever a concrete set of built-in functions is used.

Reply:
Done.